src/getters/awaitGraphBlock.ts

```diff
@@ -0,0 +1,32 @@
+import getGraphClient from '~/getters/getGraphClient'
+import {
+    GetMetadataDocument,
+    GetMetadataQuery,
+    GetMetadataQueryVariables,
+} from '~/generated/gql/network'
+
+export const awaitGraphBlock = (blockNumber: number): Promise<void> => {
+    return new Promise((resolve) => {
+        const interval = setInterval(async () => {
+            const client = getGraphClient()
+            const {
+                data: { _meta: meta },
+            } = await client.query<GetMetadataQuery, GetMetadataQueryVariables>({
+                query: GetMetadataDocument,
+            })
+
+            if (!!meta && meta.block.number >= blockNumber) {
+                clearInterval(interval)
+                resolve()
+            }
+
+            await client.clearStore()
+        }, 500)
+
+        setTimeout(() => {
+            // cancel polling after 15 seconds
+            clearInterval(interval)
+            resolve()
+        }, 15000)
+    })
+}
```

Comment:
What's the purpose of this? Await the first "next" block?

Reply:
Ah, ok, wait for the graph to sync up, eh?

Reply:
Yes, we're waiting for the graph to sync up, but max for 15 seconds